### DISCUSSION TITLE
请教Aptos Framework中的问题

### DISCUSSION THREAD

**Discussion by jackcpku** [NonceGeek/Move-Camp#173] [Q&A] [answered]
在Aptos中，账户用结构体 `0x1::account::Account` 进行抽象如下

```
    /// Resource representing an account.
    struct Account has key, store {
        authentication_key: vector<u8>,
        sequence_number: u64,
        guid_creation_num: u64,
        coin_register_events: EventHandle<CoinRegisterEvent>,
        key_rotation_events: EventHandle<KeyRotationEvent>,
        rotation_capability_offer: CapabilityOffer<RotationCapability>,
        signer_capability_offer: CapabilityOffer<SignerCapability>,
    }
```

请问其中的 `rotation_capability_offer` 和 `signer_capability_offer` 分别代表什么含义？ `CapabilityOffer` 又表示什么？

**Comment by Demian101** [comment]
mark 一下， 我最近也在努力看源码
看看过几天能不能回答

**Comment by jackcpku** [comment]
这个问题没人解答嘛😂

**Reply by leeduckgo** [reply]
哈哈，已经同步给助教。

**Comment by hqwangningbo** [comment, marked as answer]
signer_capability_offer是Account结构体下的属性，对应的数据是CapabilityOffer结构体，CapabilityOffer结构体里面又放着类型为SignerCapability的值，终究就是存个地址，这个地址是代理的地址，你把签名权限授权给一个地址，就会存到这
具体对应方法是
offer_signer_capability alice签名授权给代理账户bob
create_authorized_signer bob代理账户能使用alice做签名调用
rotation_capability_offer目前没看到是做什么
